src/NLog/Targets/ConsoleTarget.cs

```diff
@@ -230,27 +238,130 @@ protected override void Write(LogEventInfo logEvent)
                 return;
             }
 
-            WriteToOutput(RenderLogEvent(Layout, logEvent));
+            RenderToOutput(Layout, logEvent);
+        }
+
+        /// <inheritdoc/>
+        protected override void Write(IList<AsyncLogEventInfo> logEvents)
+        {
+            if (_pauseLogging)
+            {
+                return;
+            }
+
+            if (WriteBuffer)
+            {
+                var output = GetOutput();
+                using (var targetBuffer = _reusableEncodingBuffer.Allocate())
+                using (var targetBuilder = ReusableLayoutBuilder.Allocate())
+                {
+                    int environmentNewLineLength = System.Environment.NewLine.Length;
+                    int targetBufferPosition = 0;
+                    try
+                    {
+                        for (int i = 0; i < logEvents.Count; ++i)
+                        {
+                            targetBuilder.Result.Length = 0;
+                            Layout.RenderAppendBuilder(logEvents[i].LogEvent, targetBuilder.Result);
+                            if (targetBuilder.Result.Length > targetBuffer.Result.Length - targetBufferPosition - environmentNewLineLength)
+                            {
+                                if (targetBufferPosition > 0)
```

Comment:
Please split this method in multiple parts to reduce the number of lines and to reduce the nesting. 

I try to keep nesting at max of 3

Reply:
Extracted into smaller methods with less nesting.